### DISCUSSION TITLE
Week 7 Check In

### DISCUSSION THREAD

**Discussion by minhphan03** [UW-Upwelling-Project/Summer-2023#12] [General]
______________________

**Comment by minhphan03** [comment]
@eeholmes can you grant me permission to push the notebook updates to the Hackweek website repo? I could clone the repo but now I cannot push changes.

# Monday check in:
- Finished processing CMIP6 data, now in the data folder
- Zarr files for daily and monthly variables (low quality grid resolution) for all three scenarios (6 files in total)
- .nc files for each of the daily and monthly temporal resolution of the three scenarios (6 files in total)
- Add animation tutorial for graphing data notebook

Next steps: working on new tutorials for the new CMIP6 data

**Comment by NaNa7Miiii** [comment]
Monday Check in:
* Trained a model based on air temperature
* Add dropout and L2 regularization to the model architecture to prevent overfitting
* Trained and test a model based on randomly generated data (I expect the accuracy to be low thus it will prove the model is learning useful features instead of noises)

Here is how the **randomly generated data** I used for testing looks like:
<img width="630" alt="Screenshot 2023-07-24 at 3 57 55 PM" src="https://github.com/UW-Upwelling-Project/Summer-2023/assets/97627889/aaebe48b-b514-4c4b-b789-7f9f470e98f2">

Here is its accuracy after I applied **dropout and L2 regularization**:
<img width="992" alt="Screenshot 2023-07-24 at 3 59 25 PM" src="https://github.com/UW-Upwelling-Project/Summer-2023/assets/97627889/392ec6be-2642-439d-8098-45d21c25f9ff">
This is what I expected: a test accuracy of 51.42% (close to random guess, since we are dealing with a binary classification) when early stopping was triggered and a validation accuracy of 0.491 for the last batch. I have also tested the model without early stopping, it has a pretty high accuracy indicating it was learning noises after the moment it should stop. 

Here is the SST model after I applied **dropout and L2 regularization**:
<img width="992" alt="Screenshot 2023-07-24 at 4 02 44 PM" src="https://github.com/UW-Upwelling-Project/Summer-2023/assets/97627889/8126993c-6ae7-4e17-a6ca-6bcb159a0816">

Here is the air temperature model after I applied **dropout and L2 regularization**:
<img width="958" alt="Screenshot 2023-07-24 at 4 33 21 PM" src="https://github.com/UW-Upwelling-Project/Summer-2023/assets/97627889/f60e3a29-1ca2-40cc-9d55-71a766205e4f">

**Comment by minhphan03** [comment]
Minh's Tuesday check-in:
- Work on 2 tutorials on the CMIP6
- Resolved issue with NOAA erddap

**Comment by NaNa7Miiii** [comment]
Tuesday Check-In:
* Generating new label .csv file using Minh's zarr file instead
* model interpretation

**Reply by NaNa7Miiii** [reply]
![Uploading Screenshot 2023-07-25 at 4.10.52 PM.png…]()

**Comment by minhphan03** [comment]
Indian Ocean warming to trends of Arctic Oscillation (idea: https://www.nature.com/articles/s41612-022-00279-x):

* "Momentum flux convergence by stationary waves due to the Indian Ocean warming plays positive trend of arctic oscillation, shown by a poleward shift of zonal-mean zonal winds"
* "...reduced propagating wave activity flux over North Pacific (Ocean?) due to warming also plays a role in polar vortex strengthening..."


Eli: Can also look at North Atlantic Oscillation and the Indian Ocean Dipole. Ask if + and - AO and NAO values are associated with different patterns of SST or wind or SSH (sea level anomaly) are associated. So rather than looking at average SST across the Arabian Sea, use image classification (or image clustering) to look at whether + or - AO (NAO) are associated with particular patterns.

# Developing ideas from this:
- Can we create a model to predict the pattern and inter-relationship between the two phenomenon? NAO index (nasa.gov) monthly data can be obtained, as well as indian ocean dipole index (https://psl.noaa.gov/gcos_wgsp/Timeseries/DMI/) time series can help us bring an interesting viewpoint to our dataset.

**Comment by minhphan03** [comment]
# New Idea: Investigating Garbage /Plastic concentration in indian ocean & relation to upwelling?:
- monthly data https://podaac.jpl.nasa.gov/dataset/CYGNSS_L3_MICROPLASTIC_V1.0
- paper: https://www.sciencedirect.com/science/article/pii/S0025326X23002692#da0005 & https://microplastics.springeropen.com/articles/10.1186/s43591-021-00013-z#availability-of-data-and-materials

**Comment by NaNa7Miiii** [comment]
Jiarui Wed & Thurs:
Project Pitch:
I'm planning to develop a model specifically tailored to predict Sea Surface Temperature (SST). Differing from traditional image classification models, the output of our model isn't a category or probability, but a forecasted SST image consistent in size with the input. The implications of SST are vast, impacting areas like climate change, weather forecasting, and marine ecosystems. I possess a Zarr dataset of ocean variables from 1979-2022 for the Indian Ocean and Bay of Bengal, which includes SST data. Within this region, SST fluctuations are quite dynamic, especially during summer months where the contrast between near-shore and off-shore temperatures is prominent. This characteristic will aid our model in effectively learning valuable features. Participants will gain hands-on experience and refine their skills in handling large datasets, understanding regional climate variations, and applying machine learning techniques. 

* Research in the forecast model architecture

**Comment by NaNa7Miiii** [comment]
Jiarui Friday Checkin:
* ML tutorials: the machine learning tutorials are in the folder: shared/jiarui-notebooks/ml_tutorials. 
* Study Transformers in image classification / next-frame prediction
